src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -570,61 +612,95 @@ private async Task ProcessSubscriptionAsync()
+                                    subscriptionChangeVectorBeforeCurrentBatch = _lastChangeVector ?? SubscriptionState.ChangeVectorForNextBatchStartingPoint;
 
-                                if (sendingCurrentBatchStopwatch.ElapsedMilliseconds > 1000)
-                                    await SendHeartBeat("Didn't find any documents to send and more then 1000ms passed");
+                                    updateBatchPerformanceStats();
+                                    
+                                    if (sendingCurrentBatchStopwatch.ElapsedMilliseconds > 1000)
+                                        await SendHeartBeat("Didn't find any documents to send and more then 1000ms passed");
+                                    
+                                    using (docsContext.OpenReadTransaction())
+                                    {
+                                        var globalEtag = useRevisions
+                                            ? TcpConnection.DocumentDatabase.DocumentsStorage.RevisionsStorage.GetLastRevisionEtag(docsContext, Subscription.Collection)
+                                            : TcpConnection.DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(docsContext.Transaction.InnerTransaction, Subscription.Collection);
 
-                                using (docsContext.OpenReadTransaction())
-                                {
-                                    var globalEtag = useRevisions ?
-                                        TcpConnection.DocumentDatabase.DocumentsStorage.RevisionsStorage.GetLastRevisionEtag(docsContext, Subscription.Collection) :
-                                        TcpConnection.DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(docsContext.Transaction.InnerTransaction, Subscription.Collection);
+                                        if (globalEtag > _startEtag)
+                                            continue;
+                                    }
 
-                                    if (globalEtag > _startEtag)
+                                    AssertCloseWhenNoDocsLeft();
+                                    
+                                    if (await WaitForChangedDocuments(replyFromClientTask)) 
                                         continue;
                                 }
+                            }
 
-                                AssertCloseWhenNoDocsLeft();
-
-                                if (await WaitForChangedDocuments(replyFromClientTask))
-                                    continue;
+                            // using (batchParentScope.For(SubscriptionBatchOperations.AckScope))
+                            using (batchParentScope.For(SubscriptionOperationScope.Ack))
+                            {
+                                batchParentScope.RecordStartWaitingForClientAck(DateTime.UtcNow);
```

Comment:
Don;t use DateTime.UtcNow in Raven code. Look at SystemTime instead.

Reply:
Must RecordStartWaitingForClientAck method be so dump? ;-) Let's grab current date inside - here we can just say: RecordStartWaitingForClientAck().

Reply:
done